packages/cadl-ranch-specs/http/azure/core/main.cadl

```diff
@@ -117,29 +138,65 @@ op get is ResourceOperations.ResourceRead<User>;
 @doc("Lists all Users")
 @summary("Lists all users.")
 @scenarioDoc("""
-Should only generate one model named User.
+Should only generate models named User and UserOrder.
 
 Should not generate visible model like CustomPage.
 
-Expected query parameter: api-version=2022-12-01-preview
+Expected query parameter: api-version=2022-12-01-preview&top=5&skip=10&orderby=id&filter=id%20eq%201&select=id&select=orders&select=ETag&expand=orders
 
 Expected response body:
 ```json
 {
   "value":[
      {
         "id":1,
-        "name":"Madge"
+        "name":"Madge",
+        "ETag": "11bdc430-65e8-45ad-81d9-8ffa60d55b59",
+        "orders": [{ "id": 1, "userId": 1, detail: "a recorder" }]
      },
      {
         "id":2,
-        "name":"John"
+        "name":"John",
+        "ETag": "11bdc430-65e8-45ad-81d9-8ffa60d55b5a",
+        "orders": [{ "id": 2, "userId": 2, "detail": "a TV" }]
      }
   ]
 }
 ```
 """)
-op list is ResourceOperations.ResourceList<User>;
+op list is ResourceOperations.ResourceList<
+  User,
+  ListQueryParametersTrait<global.Azure.Core.StandardListQueryParameters &
+    global.Azure.Core.OrderByQueryParameter &
+    global.Azure.Core.FilterQueryParameter &
+    global.Azure.Core.SelectQueryParameter &
+    global.Azure.Core.ExpandQueryParameter>
```

Comment:
We may need to double check whether they should be csv or multi.

Not blocking this (just behavior may change).

Reply:
checked the compiler export them as multi:
![image](https://user-images.githubusercontent.com/59815250/222335651-0fbf0317-50c8-4e63-b44a-cea2e85f2ce3.png)

Reply:
I know, and that likely to be wrong.
https://www.odata.org/getting-started/basic-tutorial/